### PR TITLE
Addeed get_friend_name() threading tests

### PR DESCRIPTION
**Pull request by ilyachur**
### Tickets:
 - 57069

### CODE REVIEW

File: ngraph/test/threading.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    std::vector<std::thread> threads(number);
+
+    for (size_t i = 0; i < threads.size(); i++)
+        threads[i] = std::thread(get_friendly_name, f);
+
+    for (auto& th : threads)
+    {
+        th.join();
+    }
```
Comment (ilya-lavrenov) [inline ngraph/test/threading.cpp:51]: looks like we need to check that name is the same in all threads

Comment (ilyachur) [reply ngraph/test/threading.cpp:51]: It is not a final PR 😄 I cannot reproduce the issue locally and created this PR to reproduce it in our CI.

Comment (ilya-lavrenov) [reply ngraph/test/threading.cpp:51]: but what are you trying to reproduce? you don't have any asserts in tests that `names` are the same for all threads. Probably, we need to create functions with more nodes while this one is quite short

Comment (ilyachur) [reply ngraph/test/threading.cpp:51]: We had an issue, that CPU plugin got the crash on centOS from this method.

Comment (ilya-lavrenov) [reply ngraph/test/threading.cpp:51]: I suppose crash happens not inside of this method, it happens as a result of different names in different CPU streams.

Comment (ilyachur) [reply ngraph/test/threading.cpp:51]: Yeah, but I check all names on my laptop  and they are the same. I will try to add additional checks.

**Review by ilya-lavrenov — APPROVED** [review]

File: ngraph/core/include/openvino/core/node.hpp [hunk 2]
```diff
@@ -513,6 +514,7 @@ class OPENVINO_API Node : public std::enable_shared_from_this<Node> {
     std::shared_ptr<ngraph::op::util::OpAnnotations> m_op_annotations;
     OPENVINO_SUPPRESS_DEPRECATED_END
     std::map<std::string, std::shared_ptr<Variant>> m_rt_info;
+    mutable std::mutex m_mutex;
```
Comment (myshevts) [inline ngraph/core/include/openvino/core/node.hpp:517]: what if we have thousands of nodes- that would be too bad (afaik only 16M mutexes are allowed in Linux)... also having a mutex per node seems to be VERY redundant

**Pull request merged**